packages/components/src/__actions__/Button/v3/subcomponents/ButtonContent/ButtonContent.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react"
+import classNames from "classnames"
+import { VisuallyHidden } from "~components/VisuallyHidden"
+import { ButtonProps } from "../../Button"
+import { PendingButtonProps } from "../../types"
+import styles from "./ButtonContent.module.css"
+
+type ButtonContentProps = {
+  children?: React.ReactNode
+  size?: ButtonProps["size"]
+  icon?: ButtonProps["icon"]
+  iconPosition?: ButtonProps["iconPosition"]
+  hasHiddenLabel?: ButtonProps["hasHiddenLabel"]
+} & Pick<PendingButtonProps, "isPending" | "hasHiddenPendingLabel">
+
+/** Renders the inner content of the button, handling icon and label visibility during pending state */
+export const ButtonContent = ({
+  children,
+  hasHiddenLabel,
+  size = "medium",
+  icon,
+  iconPosition = "start",
+  isPending,
+  hasHiddenPendingLabel,
+}: ButtonContentProps): JSX.Element => (
+  <span
+    className={classNames(
+      styles.buttonContent,
+      styles[size],
+      isPending && hasHiddenPendingLabel
+        ? styles.buttonContentHidden
+        : isPending && styles.buttonContentPending
+    )}
+    aria-hidden={isPending}
```

Comment:
I think you can pass these in from the parent instead. Then this way, this component doesn't need to know anything about pending states, it is simply controlling rendering the non-pending content :)

Reply:
📁 ◀️ 🚚  Have shifted them into button and passed them in through className